### PR TITLE
CI Pipelines build with WDK Nuget Packages

### PR DESCRIPTION
**Pull request by middlemose**
With the WDK now supporting Nuget packages for ease of install it is beneficial to have the CI pipeline also utilize this ability. This will allow easier and faster integration when new WDK versions are published without relying on the runner images to be updated.

- Adds Nuget package capability to the CI pipeline.
- Adds installation of WDK vsix directly into the CI pipeline.
- Updates environment variables for ease of incrementing component versions.

### Pipeline Time Metrics
**BuildAll** (push and pull_request)
Without Vsix Install: ~08m 59s
With Vsix Install: ~38m 59s (optimized ~8m)

**CodeQL** (push and pull_request)
Without Vsix Install: ~1h 1m 43s
With Vsix Install: ~1h 34m 31s (optimized ~1h 16m 2s)

**Average** WDK Vsix Install Time
~21m (optimized ~15s)

**Average** WDK Nuget Restore
~1m 30s

### CODE REVIEW

File: .github/scripts/Install-Vsix.ps1 [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+[CmdletBinding()]
+param(
+    [bool]$optimize = $false
+)
+
+$root = Get-Location
+
+# launch developer powershell (if necessary)
+if (-not $env:VSCMD_VER) {
```
Comment (JakobL-MSFT) [inline .github/scripts/Install-Vsix.ps1:16]: Sorry to be late here, but I would imagine that within GitHub either ($env:VSCMD_VER) is true or it is false.  Always.  Why do you have to code up a conditional manner?  Why not just assert what you believe to be true?

File: Build-SampleSet.ps1 [hunk 2]
```diff
@@ -9,6 +9,17 @@ param(
 )
 
 $root = Get-Location
+
+# launch developer powershell (if necessary)
```
Comment (JakobL-MSFT) [inline Build-SampleSet.ps1:13]: Similar to a comment for Install-Vsix.ps1.  Sorry to be late here, but I would imagine that anywhere (developer command prompt or GitHub either ($env:VSCMD_VER) is expected to be true or it is expected to be false. Why do you have to code up a conditional manner? Why not just assert what you believe to be true?

**Review by JakobL-MSFT — APPROVED** [review]
Approved with non-blocking comments.

File: Build-SampleSet.ps1 (hunk 2 above)
Comment (middlemose) [reply Build-SampleSet.ps1:13]: The thought behind this logic is when a user executes the script(s) locally and the environment may or may not be a developer shell already.

File: .github/scripts/Install-Vsix.ps1 (hunk 1 above)
Comment (middlemose) [reply .github/scripts/Install-Vsix.ps1:16]: Thanks for comment here. Will consider changing in the future.

**Pull request merged**